extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetSchemaReader.java

```diff
@@ -143,18 +147,35 @@ public static ParquetInstructions readParquetSchema(
         };
         final ParquetMessageDefinition colDef = new ParquetMessageDefinition();
         final Map<String, String[]> parquetColumnNameToFirstPath = new HashMap<>();
-        for (final ColumnDescriptor column : schema.getColumns()) {
+        final Map<Integer, Long> fieldIdCount = schema.getColumns()
+                .stream()
+                .map(ColumnDescriptor::getPrimitiveType)
```

Comment:
I find it weird that field ID is on the primitive type.

Reply:
Good catch, yes; in the case of a list type, the field id is on the list and not on the primitive.